src/sysroots.c

```diff
@@ -215,12 +202,5 @@ void SySetGapRootPath(const Char * string)
 */
 Obj SyGetGapRootPaths(void)
 {
-    Obj tmp = NEW_PLIST_IMM(T_PLIST, MAX_GAP_DIRS);
-    for (int i = 0; i < MAX_GAP_DIRS; i++) {
-        if (SyGapRootPaths[i][0]) {
-            PushPlist(tmp, MakeImmString(SyGapRootPaths[i]));
-        }
-    }
-    MakeImmutableNoRecurse(tmp);
-    return tmp;
+    return SyGapRootPaths;
```

Comment:
Actually this change already ensures that the library has access to the kernel root paths via `GAPInfo.KernelInfo.GAP_ROOT_PATHS`.

We then do `GAPInfo.RootPaths:= GAPInfo.KernelInfo.GAP_ROOT_PATHS;` in `system.g` which is fine.

But in `ExtendRootDirectories` we do this:
```gap
      GAPInfo.RootPaths:= Immutable( Concatenation( GAPInfo.RootPaths,
          rootpaths ) );
```
which breaks the connection. So this function needs to be changed to instead modify the plist "in place". For this the list must not be made immutable (indeed that's why I removed the code doing that here).